src/lib.rs

```diff
@@ -20,6 +25,13 @@ pub unsafe fn start(pointer: *mut u8, length: usize) {
 
 #[cfg(target_arch = "wasm32")]
 #[js_export]
-fn start(rom: Vec<u8>) {
-    gameboy_opengl_web::start(rom);
+pub fn start(rom: Vec<u8>, dom_ids: String) {
+    let res = serde_json::from_str(dom_ids.as_str());
+    if res.is_ok() {
+        gameboy_opengl_web::start(rom, res.ok().unwrap());
```

Comment:
I think this code could get improved in two ways. 1) we can just call `res.unwrap()`, no need to turn it into an Option with `ok()`. 2) Let's pass the error message up to the console by calling `res.unwrap_err()` and formatting it. You will need to derive `Debug` for the `DOMInfo` struct. Also, I don't think the `JsSerialize` import is used anywhere.

```rust
#[cfg(target_arch = "wasm32")]
#[js_export]
pub fn start(rom: Vec<u8>, dom_ids: String) {
    let res = serde_json::from_str(dom_ids.as_str());
    if res.is_ok() {
        gameboy_opengl_web::start(rom, res.unwrap());
    } else {
        let msg = format!("{:?}", res.unwrap_err());
        console!(log, "bad parse: ", msg);
    }
}
```

Reply:
Actually, using pattern matching makes it even cleaner

```rust
#[cfg(target_arch = "wasm32")]
#[js_export]
pub fn start(rom: Vec<u8>, dom_ids: String) {
    match serde_json::from_str(dom_ids.as_str()) {
        Ok(dom_info) => gameboy_opengl_web::start(rom, dom_info)
        Err(err) =>  {
            let msg = format!("{:?}", err);
            console!(log, "bad parse: ", msg);
        }
    };
}
```

Reply:
I added those changes. It should be cleaner now.